extensions/amp-mathml/1.0/component.js

```diff
@@ -0,0 +1,81 @@
+import {MessageType, deserializeMessage} from '#core/3p-frame-messaging';
+import {px} from '#core/dom/style';
+import {tryParseJson} from '#core/types/object/json';
+
+import * as Preact from '#preact';
+import {useCallback, useMemo, useState} from '#preact';
+import {forwardRef} from '#preact/compat';
+import {ProxyIframeEmbed} from '#preact/component/3p-frame';
+
+import {TYPE} from './utils';
+
+const INLINE_STYLES = {
+  display: 'inline-block',
+  'vertical-align': 'middle',
+};
+
+const ORIGIN_MATCHER = () => true;
+
+/**
+ * @param {!BentoMathmlDef.Props} props
+ * @param {{current: BentoMathmlDef.Api}} ref
+ * @return {PreactDef.Renderable}
+ */
+function BentoMathmlWithRef(
+  {
+    formula,
+    inline = false,
+    title = 'MathML formula',
+    style: bentoMathmlStyles = {},
+    onLoad,
+    ...rest
+  },
+  ref
+) {
+  const [dimensions, setDimensions] = useState(null);
+
+  const messageHandler = useCallback(
+    (e) => {
+      const data = tryParseJson(e.data) ?? deserializeMessage(e.data);
+      if (data['type'] == MessageType.EMBED_SIZE) {
+        const {height, width} = data;
+        setDimensions({
+          height: px(height),
+          width: inline ? px(width) : undefined,
+        });
```

Comment:
Let's also use the `useIntersectionObserver` so that any dimension changes occur out of viewport (CLS compliant)

Reply:
implemented Intersection Observer logic - hides formula until it is offscreen, then sets the dimensions.